cabal-install/Distribution/Client/ProjectBuilding.hs

```diff
@@ -593,7 +596,7 @@ data BuildResult = BuildResult {
        buildResultDocs    :: DocsResult,
        buildResultTests   :: TestsResult,
        buildResultLogFile :: Maybe FilePath,
-       buildResultLibInfo :: [InstalledPackageInfo]
+       buildResultLibInfo :: Maybe InstalledPackageInfo
```

Comment:
I was wondering before if we ought to be a bit more explicit either with the type or with assertions that we will have the ipkg iff the package/component has/is a lib.